src/models/engine_models.rs

```diff
@@ -24,6 +26,9 @@ pub enum EngineError {
 impl fmt::Display for EngineError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
+            EngineError::EngineNotFound => {
+                write!(f, "Search engine not found")
+            }
```

Comment:
```suggestion
            EngineError::EngineNotFound(engine) => {
                write!(f, "No such engine with the name '{engine}' found")
            }
```

I would like to suggest another small change. Could you make the error message a little more verbose like including the engine name because this will help the user spot the issue more quickly and easily. :slightly_smiling_face:

Reply:
Yes, that's a good point!